python/src/aac/aac.aac

```diff
@@ -0,0 +1,1265 @@
+    'extends' field of another schema wishing to inherit from this schema.
+---
+modifier:
+  name: final
+  description: |
+    Denotes that the schema is final and cannot be referenced by another
+    schema's 'extends' field for inheritance.
+---
+schema:
+  name: Enum
+  package: aac.lang
+  extends: 
+    - package: aac.lang
+      name: AacType
+  root: enum
+  description: |
+    A definition that represents an enumerated type of value.
+    
+    An example of when to use an 'enum' is when you want to provide several
+    different options for a single value.
+  fields:
+    - name: values
+      type: string[]
+      description: |
+        A list of strings that encompass the enumeration's possible values.
+      is_required: true
+---
+schema:
+  name: Primitive
+  package: aac.lang
+  extends: 
+    - package: aac.lang
+      name: AacType
+  root: primitive
+  description: |
+    A definition that represents a primitive value for use in the model.
+  fields:
+    - name: python_type
+      type: string
+      description:  The type value to represend this primitive with while generating python code.
+      is_required: true
+    - name: constraints
+      type: PrimitiveConstraintAssignment[]
+      description: |
+        A list of constraints to apply to the primitive.
+---
+primitive:
+  name: string
+  package: aac.lang
+  description:  |
+    A textual value.
+  python_type: str
+  constraints:
+    - name: Check string
+---
+primitive:
+  name: int
+  package: aac.lang
+  python_type: int
+  constraints:
+    - name: Check int
+---
+primitive:
+  name: number
+  package: aac.lang
+  python_type: float
+  constraints:
+    - name: Check number
+---
+primitive:
+  name: bool
+  package: aac.lang
+  python_type: bool
+  description:  |
+    A boolean value.
+  constraints:
+    - name: Check bool
+---
+primitive:
+  name: date
+  package: aac.lang
+  python_type: str
+  description:  |
+    A date or date-time value.
+  constraints:
+    - name: Check date
+---
+primitive:
+  name: file
+  package: aac.lang
+  python_type: str
+  description:  |
+    A computer file path value.
+  constraints:
+    - name: Check file
+---
+primitive:
+  name: directory
+  package: aac.lang
+  python_type: str
+  description:  |
+    A computer directory path value.
+  constraints:
+    - name: Check directory
+---
+primitive:
+  name: any
+  package: aac.lang
+  python_type: Any
+  description:  |
+    A catch-all primitive type that will accept any value.  It is recommended to avoid the use
+    of any as it can make it difficult to reason about your model.  Due to the broad applicability of the 
+    any primitive, it is not possible to provide a general purpose constraint checker. If you find a need to use
+    the any primitive, we recommend defining a custom constraint to check the value for your usage.
+---
+primitive:
+  name: dataref
+  package: aac.lang
+  python_type: str
+  description:  |
```

Comment:
The new `dataref` and `typeref` primitives were said to be used to replace a handful of validators. So, the logic for parsing, implementing, and handling encountering these types needs to be evaluated for meeting the needs of these tasks:
- #729
- #730 
- #732 
- #735 
- #737 
- #743 
- #744

Reply:
I have not gotten to test these referencing capabilities yet to see how `check` returns on them to confirm they were brought forward as a `primitive` constraint as discussed in the deep dive.

@cvchaparro , @bunchw , and @crazynewidea if you start testing today, please try to incorporate evaluating these.

Reply:
The dataref primitive type has some tests defined in the test_check_dataref() function on line 141 of the file at :
    /workspace/AaC/python/tests/test_aac/plugins/aac_primitives/test_aac_primitive_constraints.py
I temporarily modified the check for "not_a_field_name_that_exists" on line 55 to look for SUCCESS instead of FAILURE to verify that the test is being run by the tox configuration in the workspace. It is indeed being run.

Reply:
Testing for #743  and #744 of legacy validators brought forward, the `dataref` primitive constraint handles when those situations are encountered.

```
(venv) gitpod /workspace/AaC/python/features/alarm_clock (pure-py-api) $ aac check --verbose usecase.yaml 
Dataref constraint failed for value 'users': 'users' not found in 'actor.name' values '['user', 'alarm_clock', 'timer', 'clock', 'alarms']'
  Source: /workspace/AaC/python/features/alarm_clock/usecase.yaml (26:2:344:12)
Dataref constraint failed for value 'alarm': 'alarm' not found in 'actor.name' values '['user', 'alarm_clock', 'timer', 'clock', 'alarms']'
  Source: /workspace/AaC/python/features/alarm_clock/usecase.yaml (37:2:575:6)
Dataref constraint failed for value 'setAlarms': 'setAlarms' not found in 'model.behavior.name' values '['setAlarm', 'getTime', 'setTime', 'triggerTimer', 'triggerAlarm']'
  Source: /workspace/AaC/python/features/alarm_clock/usecase.yaml (38:2:591:6)
Dataref constraint failed for value 'alarm': 'alarm' not found in 'actor.name' values '['user', 'alarm_clock', 'timer', 'clock', 'alarms']'
  Source: /workspace/AaC/python/features/alarm_clock/usecase.yaml (42:2:682:6)
Dataref constraint failed for value 'alarm': 'alarm' not found in 'actor.name' values '['user', 'alarm_clock', 'timer', 'clock', 'alarms']'
  Source: /workspace/AaC/python/features/alarm_clock/usecase.yaml (49:2:833:12)
Dataref constraint failed for value 'alarm': 'alarm' not found in 'actor.name' values '['user', 'alarm_clock', 'timer', 'clock', 'alarms']'
  Source: /workspace/AaC/python/features/alarm_clock/usecase.yaml (68:2:1247:6)
```
Thoughts @cvchaparro , @crazynewidea , @bunchw ?

Think some clean-up of the returned location, it has line number and such, maybe just more clarification on what each number is? Posted a similar comment on the portion of code that returns this information. 

Otherwise, all that is missing from those two, I believe, is the documentation on them. Documentation on the categories of constraints exist, but documentation on each of the constraints currently provided with AaC does not.

Reply:
Tested for #735 by altering the listed requirement. Again, similar opinions to the other `dataref` and `typeref` constraint returns of clean-up of returned location and missing documentation. 

```
(venv) gitpod /workspace/AaC/python/features/calc/spec ((5d6a1c3...)) $ aac check Add_SRS.yaml 
Dataref constraint failed for value 'ADD-2': 'ADD-2' not found in 'req.id' values '['ADD-1', 'SM-1', 'CALC-1', 'CALC-2', 'CALC-3', 'CALC-4', 'IM-1', 'MULT-1', 'DIV-1', 'CM-1', 'SQRT-1', 'POW-1', 'SUB-1']'
  Source: /workspace/AaC/python/features/calc/spec/Add_SRS.yaml (7:2:185:12)
```

Reply:
I am not certain how to try and test for #730 functionality, the legacy validator was "An AaC plugin that enables validating the format of a query reference." @cvchaparro , thoughts on one how to check if this is still accommodated via the `typeref` and `dataref`?

Reply:
I agree with your rationale behind completion of Issues #743, #744, and #645, and agree that the location output could use some minimal identification of the meaning of each location numbers.